___
# Exercício: Teste de hipóteses para média populacional 
___

## Aula 24

<div id="indice"></div>

## Índice

- [Exercício 1](#ex1)
- [Exercício 2](#ex2)
- [Exercício 3](#ex3)
- [Exercício 4](#ex4)
- [Respostas](#respostas)
___

In [3]:
from scipy import stats
import numpy as np
import pandas as pd
import math
import matplotlib.pyplot as plt
%matplotlib inline

<div id="ex1"></div>

### EXERCÍCIO 1

Um estudo foi desenvolvido para avaliar o salário de empregadas domésticas na cidade de São Paulo. 

Foram sorteadas e entrevistadas $121$ trabalhadoras. Admita que o desvio padrão dessa variável na cidade é de 0,88 salários mínimos.

Responda:

**a.** Você conhece a distribuição do estimador $\overline{X}$? Se não, é possível fazer alguma suposição?

**b.** Desconfia-se que a média salarial seja superior a 2 salários mínimos. Baseado nessa informação, formule as hipóteses em termos do problema e em termos do parâmetro.

**c.** Para um nível de significância de 2%, construa a RC.


**d.** Se a amostra forneceu média de $2.38$ salários mínimos, qual a conclusão deste teste?

**e.** Que suposições você fez para resolver os itens anteriores?

[Respostas](#respostas)

In [ ]:
# ESCREVA SEU CÓDIGO AQUI

In [5]:
sigma = 0.88
var = sigma**2
n = 121
#X barra é o conjunto de todas médias

In [ ]:
#a) Assumindo que 121 é um n suficiente, então o X barra possui distribuição normal com média mu e desvio padrão de 0.88/121**0.5
## X_barra ~N(mu, 0.88**2/121) #0.88**2/121 é a variancia

In [ ]:
#b) 
# Em termos do problema:
# H0: a média salarial não é superior a 2 salários mínimos
# H1: a média salarial é superior a 2 salários mínimos

# Em termos dos parametros:
# H0: 𝜇 <= 2
# H1 𝜇 < 2

In [7]:
# Região crítica é o critério de quando passa a rejeitar valores que estão acima da média, ou seja, X crítico é o ponto de corte
# e a regiao ccrítica é tudo depois dele

alpha = 0.02
mu_barra = 2 #no teste de hipotese supoe que nada mudou, ent usa a igualdade da hipotese nula
n = 121
sigma_barra = 0.88/n**0.5
x_critico = stats.norm.ppf(1-alpha, loc=mu_barra,scale=sigma_barra)
print(f' RC = \(x_obs | x_obs > {x_critico:.3f})')

 RC = \(x_obs | x_obs > 2.164)


In [8]:
# d)
x_obs = 2.38
# A direita:
if x_obs > x_critico:
    print('Rejeita H0, a média é superior a dois salários mínimos')
else:
    msg = 'Não rejeita H0, não há evidencias suficientes que a média é superior a dois salários mínimos'
    print(msg)

Rejeita H0, a média é superior a dois salários mínimos


In [ ]:
# e) Supomos que um n de 121 é suficientemente grande para aplicar o TLC e então sabemos a distribuição de X barra para fazer
# o teste


___

<div id="ex2"></div>

### EXERCÍCIO 2

A duração de pilhas falsificadas segue uma distribuição normal com média de $15 ut$ e variância $16 ut^2$. Pilhas autênticas têm uma duração média maior e mesmo desvio-padrão.  

Um lote de pilhas apreendido será leiloado e para definir seu preço é preciso decidir se são produtos falsificados ou não. 
O leiloeiro adotou como regra de decisão que se a duração média de uma amostra aleatória de $16$ pilhas for maior que $18 ut$, o lote será considerado autêntico. Critique essa regra. 

[Respostas](#respostas)

In [ ]:
# ESCREVA SUA RESPOSTA AQUI

In [ ]:
# Pilhas autênticas têm uma duração média maior e mesmo desvio-padrão.

In [9]:
# Em termos do problema:
# H0: a média de duração das pilhas é inferior ou igual a 15
# H1: a média de duração das pilhas é imaior que 15

# Em termos dos parametros:
# H0: 𝜇 <= 15 #lote falso
# H1 𝜇 > 15 #lote verdadeiro

In [16]:
# Não recebi o alpha? Então é necessário descobrir!!
n = 16
var = 16
sigma_barra = (var/n)**0.5
mu_barra = 15

# H0: mu<= 15 #lote falso
# H1: mu > 15 #lote original

alpha = 1 - stats.norm.cdf(18, loc=mu_barra, scale=sigma_barra)
alpha

0.0013498980316301035

In [17]:
print(f'{alpha*100:.4f}%')

0.1350%


In [ ]:
# Critica:
# O leiloeiro está sendo muito criterioso. A probabilidade de ele aceitar um lote falso como verdadeiro é extremamente baixa 
# (erro tipo I pequeno). Entretanto, isto diminui a probabilidade de ele identificar lotes genuinos (erro tipo II maior).


___

<div id="ex3"></div>

### EXERCÍCIO 3

Um estudo foi desenvolvido para avaliar a renda de pedreiros autônomos na cidade de São Paulo, o qual desconfia que a renda média seja superior a 3 salários mínimos. Admita que o desvio padrão dessa variável na cidade é de 0,7 salários mínimos. Foram sorteados e entrevistados 100 trabalhadores, cuja amostra forneceu média de 3,2 salários mínimos. 

**a.** Baseado nessa informação, formule as hipóteses H0 e H1 e interprete o erro do tipo I e erro do tipo II, sendo (erro I = rejeitar H0 | H0 verdade)  e (erro II = não rejeitar H0 | H0 falsa).

**b.** Qual deve ser a conclusão do estudo considerando 1% de significância?

**c.** Supondo que a hipótese nula seja verdadeira,  qual deve ser a probabilidade de observar um valor tão ou mais desfavorável à hipótese nula quanto a particular média amostral estimada. Qual a conclusão deste teste com 1% de significância?

[Respostas](#respostas)

In [ ]:
# ESCREVA SEU CÓDIGO AQUI

In [21]:
n = 100
mu = 3
sigma = 0.7/n**0.5 #se refere a uma coisa unica

In [22]:
# a)
# H0: mu<= 3 # renda média seja inferior a 3 salários mínimos
# H1: mu > 3 # renda média seja superior a 3 salários mínimos

In [23]:
# b)
alpha = 0.001
x_critico = stats.norm.ppf(1-alpha, loc=mu,scale=sigma)
print(f' RC = \(x_obs | x_obs > {x_critico:.3f})')

 RC = \(x_obs | x_obs > 3.216)


In [ ]:
# Critica: não há evidencias de que o salario tenha aumentado com nivel de significancia de 1%

___

<div id="ex4"></div>

### EXERCÍCIO 4

Em 2012, antes dos supermercados deixarem de distribuir sacolas plásticas gratuitamente para os consumidores, o consumo mensal de sacolas plásticas na cidade de São Paulo por adulto era, em média, de $25$ sacolas plásticas, com desvio padrão de $6$ unidades (valores populacionais). 

No mesmo ano, os estabelecimentos passaram a cobrar pelo uso das mesmas e depois voltaram atrás. Alguns anos se passaram e uma ONG da área de sustentabilidade deseja avaliar se essas idas e vindas deixaram o consumidor um pouco mais consciente com o meio ambiente. 

Para avaliar se consumo está mais consciente, verifique se o consumo de sacolas plásticas reduziu, em média, em mais do que $15\%$ após todo esse período, ao nível de significância de $1\%$?
A ONG monitorou, por um mês, $100$ adultos moradores da cidade e observou que o consumo médio amostral de sacolas plásticas entre eles foi de $20.01$ unidades.


[Respostas](#respostas)

In [ ]:
# ESCREVA SUA RESPOSTA AQUI

In [31]:
# Variáveis 
mu = 21.25
sigma = 6
alfa = 0.01
n = 100
mu_barra = 20.01
sigma_barra = 6/n**0.5

# H0: mu_barra >= 21.25
# H1: mu_barra < 21.25

In [32]:
# Achar o x_critico
x_critico = stats.norm.ppf(alfa, loc = mu, scale = sigma_barra)
x_critico 


19.854191275575495

In [35]:
print(f'Sempre que a media amostral for menor que {x_critico} rejeitamos a hipótese nula')
print('Portanto não rejeita-se H0')

Sempre que a media amostral for menor que 19.854191275575495 rejeitamos a hipótese nula
Portanto não rejeita-se H0


<div id="respostas"></div>

## Respostas

### [Exercício 1](#ex1)

#### b)

Hipóteses em termos do problema:
$H_0:$ a desconfiança não procede
$H_1:$ desconfiança procede

Ou seja:

$H_0: \mu \leq 2$

$H_1: \mu > 2$

#### c)

$RC=\{\overline{x}_{obs} > 2.164299912850546\}$

#### d)

Refutamos a hipótese nula

### [Exercício 2](#ex2)

A conclusão deve ser tirada a partir do cálculo do nível de significância: $\alpha=0.13498980316301035\%$

### [Exercício 3](#ex3)

#### a) 

$H_0: \mu \leq 3$ : a renda média é de 3 s.m

$H_1: \mu > 3$ : a renda média é superior a 3 s.m

Vamos adaptar o quadro de erros do tipo 1 e tipo 2 fornecido anteriormente nesta tabela

<table style="width:50%" >
<tr> <td>&nbsp;</td><td>H_0 é verdadeiro</td><td>H_0 é Falso</td></tr>
    <tr> <td rowspan=2>Rejeito H_0</td> <td rowspan=2><b>Erro to tipo 1:</b> Rejeitar que a média salarial é 3 s.m quando na verdade ela de fato é 3 s.m</td><td rowspan=2>Decisão correta</td> </tr>
<tr></tr>
<tr><td rowspan=2>Não rejeito H_0</td><td rowspan=2>Decisão correta</td><td rowspan=2><b>Erro do tipo 2:</b> Concluir que a média salarial não é maior que 3 s.m quando na verdade ela é maior</td> </tr>
<tr></tr>
</table>

#### b)

$RC =\{ {\overline{x}_{obs}>3.162844351182859} \}$

#### c)

O valor-$p$ é $0.2137366980086264\%$, que é menor do que 1\%.

### [Exercício 4](#ex4)

Hipóteses:

$H_0: \mu \geq 21.25 $

$H_1: \mu < 21.25$

Região Crítica:

$RC = \{ \overline{x}_{obs} < 19.854 \}$

Não refutamos a hipótese nula (não diga "aceitamos H0").